galaxy/api/v2/views/collection.py

```diff
@@ -143,6 +150,21 @@ def _check_namespace_access(self, namespace, user):
                 'the namespaces you have access to.'
             )
 
+    def _check_role_name_conflict(self, ns, name):
+        roles = models.Content.objects.filter(
+            content_type__name=constants.ContentType.ROLE,
+            namespace=ns,
```

Comment:
Options to consider here:
- `namespace=ns`
- `repository__provider_namespace__namespace=ns`

Reply:
We have `namespace` on the Content object for convenience, but I think to be accurate, we should use `provider_namespace__namespace`.

Reply:
update was made to `repository__provider_namespace__namespace`